validator/project_validator.go

```diff
@@ -958,6 +961,17 @@ func checkRunOn(runOnHasDistro, runOnHasContainer bool, runOn []string) []Valida
 	return nil
 }
 
+func validateProjectLimits(_ context.Context, settings *evergreen.Settings, project *model.Project, _ *model.ProjectRef, _ bool) ValidationErrors {
+	errs := ValidationErrors{}
+	if settings.TaskLimits.MaxTasksPerVersion > 0 && len(project.Tasks) > settings.TaskLimits.MaxTasksPerVersion {
```

Comment:
Will this already include all the generated tasks?

Reply:
Yes, project validation is always called on the resulting project from generate.tasks.